sklearn/metrics/cluster/supervised.py

```diff
@@ -814,6 +816,83 @@ def normalized_mutual_info_score(labels_true, labels_pred, max_n_classes=5000):
     return nmi
 
 
+def fowlkes_mallows_score(labels_true, labels_pred, max_n_classes=5000):
+    """Measure the similarity of two clusterings of a set of points.
+
+    The Fowlkes-Mallows index (FMI) is defined as the geometric mean between of
+    the precision and recall::
+
+        FMI = TP / sqrt((TP + FP) * (TP + FN))
+
+    Where :math:`TP` is the number of `True Positive` (i.e. the number of pair
+    of points that belongs in the same clusters in both labels_true and
+    labels_pred), :math:`FP` is the number of `False Positive` (i.e. the number
+    of pair of points that belongs in the same clusters in labels_true and not
+    in labels_pred) and :math:`FN`is the number of `False Negative` (i.e the
+    number of pair of points that belongs in the same clusters in labels_pred
+    and not in labels_True).
+
+    The score ranges from 0 to 1. A high value indicates a good similarity
+    between two clusters.
+
+    Parameters
+    ----------
+    labels_true : int array, shape = (n_samples,)
+        A clustering of the data into disjoint subsets.
+
+    labels_pred : array, shape = (n_samples, )
+        A clustering of the data into disjoint subsets.
+
+    max_n_classes : int, optional (default=5000)
+        Maximal number of classes handled by the adjusted_rand_score
```

Comment:
(Though I think this `max_n_classes` business is yuck, and should be avoided by using a sparse contingency matrix as in #4788)